llvm/lib/CodeGen/SelectionDAG/DAGCombiner.cpp

```diff
@@ -11440,6 +11420,28 @@ SDValue DAGCombiner::visitSRL(SDNode *N) {
   if (SDValue AVG = foldShiftToAvg(N, DL))
     return AVG;
 
+  // Fold clmul(zext(x), zext(y)) >> (BW - 1 | BW) -> clmul(r|h)(x, y).
+  if (VT.getScalarSizeInBits() % 2)
+    return SDValue();
```

Comment:
Don't early out. Inevitably someone will add code below this without noticing this. Either wrap the whole thing with an if or more move it to a function like combineShiftToMULH.